### DISCUSSION TITLE
Suggestion for new features

### DISCUSSION THREAD

**Discussion by artem59k** [BearWare/TeamTalk5#1967] [Ideas]
### Description

Hello. I suggest incorporating a security feature. I propose implementing a master password for the server administrator to authenticate important actions like channel deletion and other critical operations.
In addition, I suggest allowing users to create sub-channels within groups based on their interests. Where the channel operator can be the administrator of an interest group. Furthermore, I propose the introduction of a feature that eliminates the need for users to enter a password for private channels. Instead, they can request permission and have the channel owner or operator approve their access.
To enhance privacy, I suggest implementing a whispering function, where conversations between specific individuals are not audible to others in the channel.
For improved user experience, I recommend including user activity tracking, which displays the timestamp of a user's last login to the server. Additionally, I propose adding an auto-deletion feature for files that can be configured within the channels.
Lastly, I suggest incorporating user blocking based on device identifiers such as MAC address, disk ID, motherboard ID, or any other relevant identifiers as part of user management.

### Application

- [X] qtTeamTalk
- [X] TeamTalkAndroid
- [X] iTeamTalk
- [X] TeamTalkServer

**Comment by beqabeqa473** [comment]
lots of requests in one issue. I think discussion would be more
relevant for such type of things.

On 7/30/23, artem59k ***@***.***> wrote:
> ### Description
>
> Hello. I suggest incorporating a security feature. I propose implementing a
> master password for the server administrator to authenticate important
> actions like channel deletion and other critical operations.
> In addition, I suggest allowing users to create sub-channels within groups
> based on their interests. Where the channel operator can be the
> administrator of an interest group. Furthermore, I propose the introduction
> of a feature that eliminates the need for users to enter a password for
> private channels. Instead, they can request permission and have the channel
> owner or operator approve their access.
> To enhance privacy, I suggest implementing a whispering function, where
> conversations between specific individuals are not audible to others in the
> channel.
> For improved user experience, I recommend including user activity tracking,
> which displays the timestamp of a user's last login to the server.
> Additionally, I propose adding an auto-deletion feature for files that can
> be configured within the channels.
> Lastly, I suggest incorporating user blocking based on device identifiers
> such as MAC address, disk ID, motherboard ID, or any other relevant
> identifiers as part of user management.
>
> ### Application
>
> - [X] qtTeamTalk
> - [X] TeamTalkAndroid
> - [X] iTeamTalk
> - [X] TeamTalkServer
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/issues/1965
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by amirmahdifard** [comment]
i don't know why some people want's the teamtalk app to become so much complicated to use and stuf: why administrator should need master password, administrator is an administrator, it means they can do everything, with out any limitation: why the wispering function, if 2 users wants to speeck privately, they can send pm to each other, or they can go to a password protected channel and speeck there with voice, in faqt, we even have the ability to limit how many people can join a channel at max: i onnistly don't understand what's the point of such suggestions, when they are not needed and in faqt we already have them.

**Comment by artem59k** [comment]
There are certain circumstances when whispering in a single channel is necessary, for example, when some people like to intercept others in private channels, even if the icon is visible, and text-to-speech conversion announces that someone is intercepting. Whispering will allow you to hide the conversation and make it more confidential. A master password is needed to protect against malicious administrators who press F8 in the main channel and deliberately delete all sub-channels. It is also possible to accidentally press any object on F8 and then press the yes button. Even the most trivial example, an animal can step on F8 and then on Enter, and that's it, all hannels are deleted, or the channel that is under the cursor is deleted.

**Comment by beqabeqa473** [comment]
Such type of problems are administrators itself. Simply locking a
keyboard will save you from such things and this should be first thing
when you are going to step away from computer.

On 7/31/23, artem59k ***@***.***> wrote:
> There are certain circumstances when whispering in a single channel is
> necessary, for example, when some people like to intercept others in private
> channels, even if the icon is visible, and text-to-speech conversion
> announces that someone is intercepting. Whispering will allow you to hide
> the conversation and make it more confidential. A master password is needed
> to protect against malicious administrators who press F8 in the main channel
> and deliberately delete all sub-channels. It is also possible to
> accidentally press any object on F8 and then press the yes button. Even the
> most trivial example, an animal can step on F8 and then on Enter, and that's
> it, all hannels are deleted, or the channel that is under the cursor is
> deleted.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/issues/1965#issuecomment-1658076663
> You are receiving this because you commented.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by amirmahdifard** [comment]
@artem59k about the master password, @beqabeqa473 has answered you, and i also think that it's true: if your cat or your dog deletes all of the channels, it's not the teamtalk's problem, it is simply your problem: press nvda + 1 to lock your keyboard, or go to the desktop window and don't keep your focus on the teamtalk window.
about the wispering function you said, it is also useless, since teamtalk for windows now announces if someone intercepts you, so you can punish that person or stop your private conversation as soon as someone started intercepted you: we can only do one thing that makes sence for me more than all: also announce the interception for you in the mobile clients too: @beqabeqa473 in the windows since 5.7.1, you get alurted when someone intercepts you: could you please do the same for the mobile clients too, so we can 100% garantee people that their conversations are safe, so they won't come and ask for a wispering function, just because they are worryed about the interceptions? thanks.

**Comment by muamel1** [comment]
Talking about Whisper, I was considering suggesting it, but it seems someone already did.
However, there are always people who like to criticize even if they don't understand the suggestion. Whisper is entirely different from Intercept; I am against having Intercept because it can be used for spying in many cases. On the other hand, Whisper is when you are in a channel with many people but want to tell someone something without others hearing you, like private messages but with voice.
Moreover, this feature will help me implement an idea I already have for creating a translator. It would allow you to communicate with people who don't understand your language and many other things.

**Comment by amirmahdifard** [comment]
in faqt, i like intercepting feature, because i can use it in a lot of pozitiv ways, raather than hearing people's private conversations.
